Core/Core/Credentials/AccountManager.cs

```diff
@@ -259,7 +259,9 @@ public static void UpgradeAccount(string id)
     // setting the id to null will force it to be recreated
     account.id = null;
 
-    s_accountStorage.UpdateObject(account.id!, JsonConvert.SerializeObject(account));
+    RemoveAccount(id);
+    s_accountStorage.SaveObject(account.id, JsonConvert.SerializeObject(account));
+    await s_accountStorage.WriteComplete().ConfigureAwait(false);
```

Comment:
Seems fine, just unsure about this change to async and why WriteComplete() is necessary, seems a bit janky to me, but that's probably more about the storage?